test/std/algorithms/alg.sorting/alg.binary.search/binary.search/binary_search.pass.cpp

```diff
@@ -0,0 +1,103 @@
+// <algorithm>
+
+// template<ForwardIterator Iter, class T>
+//   constexpr bool      // constexpr after C++17
+//   binary_search(Iter first, Iter last, const T& value);
+
+#include "oneapi_std_test_config.h"
+#include "test_iterators.h"
+#include "checkData.h"
+
+#include <iostream>
+
+#ifdef USE_ONEAPI_STD
+#    include _ONEAPI_STD_TEST_HEADER(algorithm)
+#    include _ONEAPI_STD_TEST_HEADER(iterator)
+namespace s = oneapi_cpp_ns;
+#else
+#    include <algorithm>
+#    include <iterator>
+namespace s = std;
+#endif
+
+#if TEST_DPCPP_BACKEND_PRESENT
+constexpr sycl::access::mode sycl_read = sycl::access::mode::read;
+constexpr sycl::access::mode sycl_write = sycl::access::mode::write;
+
+template <class Iter, class T>
+bool
+test(Iter first, Iter last, const T& value, bool x)
+{
+    return (s::binary_search(first, last, value) == x);
+}
+
+class KernelBSearchTest1;
+class KernelBSearchTest2;
+class KernelBSearchTest3;
+class KernelBSearchTest4;
+
+template <typename Iter, typename KC>
+void
+kernel_test()
+{
+    sycl::queue deviceQueue = TestUtils::get_test_queue();
+    sycl::cl_bool ret = false;
+    sycl::range<1> numOfItems{1};
+    sycl::buffer<sycl::cl_bool, 1> buffer1(&ret, numOfItems);
+    const unsigned N = 1000;
+    const int M = 10;
+    int host_vbuf[N];
+    int x = 0;
+    for (size_t i = 0; i < N; ++i)
```

Comment:
Probably, it makes sense to make a sequence generator more advanced? 
F.e.
```
std::default_random_engine gen{std::random_device{}()};
std::uniform_real_distribution<float> dist(0, 100.0);
std::generate(in_host.begin(), in_host.end(), [&]{ return dist(gen); });
```

Reply:
@MikeDvorskiy it's from source code of importing test. We have the goal to import all these tests into our code but no goal to improve these tests and so on.

Reply:
If the test is only intended to check that a standard algorithm can be used on the device, any data are suitable I believe, and we can just generate a sorted sequence of integers as I suggested in another comment. This approach assumes that the std library implementation is well tested on its own, and we do not change it.

A different approach is to assume less - for example, expect that the implementation has different code paths, and not all might work on the device. In this case, we'd need comprehensive testing, randomization, checking special cases, etc. 

I tend to prefer the first approach, but we can discuss.